ElectronicObserver/Window/Wpf/Compass/CompassViewModel.cs

```diff
@@ -550,6 +546,28 @@ private void NextEnemyFleetCandidate(int offset)
 		*/
 	}
 
+	private List<EnemyFleetElementViewModel>? GetAllCandidateFleetViewModels()
+		=> _enemyFleetCandidate?
+				.GroupBy(f => f, new EnemyFleetElementComparer())
+				.Select(f => new EnemyFleetElementViewModel
+				{
+					EnemyFleetCandidate = f.First(),
+					Formations = f.Select(fleet => fleet.Formation).ToList()
+				})
+				.ToList();
+
+	private List<EnemyFleetElementViewModel> GetFilteredCandidateFleetViewModels()
+		=> GetAllCandidateFleetViewModels() switch
+		{
+			List<EnemyFleetElementViewModel> fleets =>
+				fleets.Where(vm => Utility.Configuration.Config.FormCompass.DisplayAllEnemyCompositions || vm.IsPreviewed).ToList() switch
+				{
+					{ Count: > 0 } fleetsFiltered => fleetsFiltered,
+					_ => fleets,
```

Comment:
The logic here should probably be...
if the setting to display all fleets is enabled => show all fleets
if the filtered list is empty => create a fleet from preview
otherwise display the filtered list

Reply:
Actually, even if display all fleets is enabled, we should probably check if the preview fleet exists in the list, and add it if not.